src/client/routes/profile/ProfileView.tsx

```diff
@@ -1,45 +1,92 @@
 import * as React from "react";
-import { withRouter } from "react-router-dom";
-import TextField from "@material-ui/core/TextField";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+import {TextField, WithStyles, withStyles, createStyles, Theme, Button, Typography} from "@material-ui/core";
 import ApiService from "../../common/ApiService";
 
-// TODO an actual view with some reasonable ui and more fields! email! password! whoa, very much data, so doge
-class ProfileView extends React.Component<any, any> {
-  constructor(props: any) {
+class ProfileView extends React.Component<IProfileViewProps, IProfileViewState> {
+  private submitField: any = React.createRef();
+
+  constructor(props: IProfileViewProps) {
     super(props);
+    this.state = {} as IProfileViewState
+  }
+
+  public async componentDidMount() {
+    const user = await ApiService.read.user();
+    this.setState(user);

```

Comment:
This replaces the whole state with the user object. I think it would be better to spread the user to state, or even better just pick the pieces we want to use here.
```js
this.setState({ ...user });
// or
this.setState({
  name: user.name,
  email: user.email
});
```